### ISSUE TITLE
Empty line break with new word breaking in v5.1.0

### ISSUE THREAD

**Issue by glitchroy** [JujuAdams/scribble#17]
The new wrapping behaviour in 5.1.0 sometimes adds an empty line break to the beginning of the string. See this example:
![](https://user-images.githubusercontent.com/31043192/67160172-46a92680-f34e-11e9-8992-e03f9b3c8238.png)

This  is the relevant code (simplified):
```js
// Create scribble structure
// ...
scribble_draw_set_wrap(-1, 120);
scribble_element = scribble_draw(0, 0, content);
    
scribble_autotype_fade_in(scribble_element, SCRIBBLE_AUTOTYPE_PER_CHARACTER, tw_speed, 0);
    
scribble_draw_reset();
```

`scribble_get_bbox(scribble_element, 0, 0)` output for string "123456789ABCDEFG" is `[ 0, 0, 123, 0, 0, 32, 123, 32 ]` for the per-word version and `[ 0, 0, 116, 0, 0, 32, 116, 32 ]` for the per-character version. The `content` variable however does not have the line break, making me think this has something to do with the new word wrapping and the max width parameter.

**Comment by JujuAdams** [comment]
Heya, thanks for reporting this bug (which pictures too!). The word wrapping stuff got messed up during optimisation when preparing 5.0.0 so I'm still tracking down the quirks.

I've got an idea what this is, and the fix should be relatively simple, but it'll be a few days before I can look at it. I am stretched very thin right now!

What's happening is that the parser is creating a newline behind the word despite that looking odd. The solution is to check if the first character of the word is also the first character on the line, and if so break the word in the middle to create a newline.

You can play with this yourself if you're feeling impatient. Line 793 in `scribble_draw()` is where newlines are handled. There's a ds_list called `_line_break_list`. At line 812, `_tell_a` is set. If `_tell_a` is equal to the last element in `_line_break_list` (before `_tell_a` is added to the list) then the block of code starting at line 815 should be ignored.

I *think* that'll solve the problem anyway. Let me know how you get on.

**Comment by JujuAdams** [comment]
Fixed in 5.1.1
